redbot/core/commands/commands.py

```diff
@@ -106,6 +107,37 @@ def parents(self):
             # We should expose anything which might be a bug in the converter
             raise exc
 
+    async def can_see(self, ctx: "Context"):
+        """Check if this command is visible in the given context.
+
+        In short, this tries to emulate the behaviour of the help formatter
+        to see if this command would be displayed when the help command is
+        invoked in the given context.
+
+        Parameters
+        ----------
+        ctx : `Context`
+            The invocation context to check with.
+
+        Returns
+        -------
+        bool
+            ``True`` if this command is visible in the given context.
+
+        """
+        for cmd in itertools.chain((self,), self.parents):
+            if cmd.hidden:
+                return False
+            try:
+                can_run = await self.can_run(ctx)
+            except commands.CheckFailure:
+                return False
+            else:
+                if can_run is False:
+                    return False
+
+        return True
```

Comment:
This logic only holds up with the default help formatter behavior, which can be changed while the bot is running by cogs. Do we care about this, or should we query the behavior of how the help formatter handles check failures?

Reply:
I'm happy with `can_see` being a simple combination of check verification and `hidden`, however I am happy to make that more clear in the docstring. This method could yet see a little tweak in the upcoming permissions overhaul too.

Reply:
I guess that's fair enough. Anything calling it should be aware that if the formatter has been told to display all visible commands, this loses meaning, and a doc string change is enough to signal that.